envoy/network/socket.h

```diff
@@ -85,6 +85,12 @@ class ConnectionInfoProvider {
    **/
   virtual absl::optional<uint64_t> connectionID() const PURE;
 
+  /**
+   * @return the name of the network interface used by the downstream connection, or unset if not
+   *available.
+   **/
+  virtual absl::optional<std::string> interfaceName() const PURE;
```

Comment:
will `absl::optional<absl::string_view>` be better to reduce unnecessary memory allocation and copy?